### PR TITLE
feat: #85 Create input component

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/component/input/Input.module.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+@import "src/styles/variables.scss";
+
+.input {
+  width: $input-width;
```
Comment (tritonJS826) [inline mw-webapp/src/component/input/Input.module.scss:4]: strange global variable

File: mw-webapp/src/component/input/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  required?: boolean;
+
+  /**
+   * Automatically focused on input after rendering if true
+   * @default false
+   */
+  autoFocus?: boolean;
+
+  /**
+   * The event handler
```
Comment (tritonJS826) [inline mw-webapp/src/component/input/Input.tsx:57]: it does not handle events

File: mw-webapp/src/component/input/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  placeholder?: string;
+
+  /**
+   * Input's class
+   */
+  className?: string | Array<string>;
+
+  /**
+   * Input's mode (defines what kind of input mode browser should present to the user)
+   * @default InputMode.text
```
Comment (tritonJS826) [inline mw-webapp/src/component/input/Input.tsx:34]: why not to use link here

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/component/input/Input.module.scss (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/input/Input.module.scss:4]: Changed to width-primary

File: mw-webapp/src/component/input/Input.tsx (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/input/Input.tsx]: Changed

File: mw-webapp/src/component/input/Input.tsx (hunk 3 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/input/Input.tsx]: ADded

File: mw-webapp/src/component/table/LoadingContext.tsx [hunk 4]
```diff
@@ -1,15 +1,25 @@
 import React, {createContext, PropsWithChildren, useContext, useState} from "react";
```
Comment (SergioAJS) [inline mw-webapp/src/component/table/LoadingContext.tsx:1]: Do we need to explicitly import react?

File: mw-webapp/src/component/input/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import React, {HTMLInputTypeAttribute} from "react";
```
Comment (SergioAJS) [inline mw-webapp/src/component/input/Input.tsx:1]: Do we need to explicitly import react?

**Review by Howl404 — APPROVED** [review]

File: mw-webapp/src/component/table/LoadingContext.tsx (hunk 4 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/table/LoadingContext.tsx]: Deleted in all components, thanks

File: mw-webapp/src/component/input/Input.tsx (hunk 5 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/input/Input.tsx]: Deleted in components, thanks

**Review by SergioAJS — APPROVED** [review]

File: mw-webapp/src/component/input/Input.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+   * Tracks the value entered into the input
+   */
+  onChange: (value: string) => void;
+
+}
+
+/**
+ * Input component
+ */
+export const Input: React.FC<InputProps> = (props: InputProps) => {
```
Comment (Howl404) [inline mw-webapp/src/component/input/Input.tsx:66]:
```suggestion
export const Input = (props: InputProps) => {
```

File: mw-webapp/src/component/table/LoadingContext.tsx [hunk 7]
```diff
@@ -1,25 +1,36 @@
...
 const LoadingContext = createContext<LoadingContextType | undefined>(undefined);
 
-type LoadingProviderProps = PropsWithChildren<unknown>;
-
-
-export const LoadingProvider: React.FC<LoadingProviderProps> = ({children}) => {
+/**
+ * Loading provider
+ */
+export const LoadingProvider: React.FC<PropsWithChildren> = (props: PropsWithChildren) => {
```
Comment (Howl404) [inline mw-webapp/src/component/table/LoadingContext.tsx:21]:
```suggestion
export const LoadingProvider = (props: PropsWithChildren) => {
```

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/table/LoadingContext.tsx (hunk 7 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/table/LoadingContext.tsx:21]: Why

File: mw-webapp/src/component/input/Input.tsx (hunk 6 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/input/Input.tsx:66]: Why

Comment (Howl404) [reply mw-webapp/src/component/input/Input.tsx:66]: Unnecessary type

Comment (Ekaterina1994) [reply mw-webapp/src/component/input/Input.tsx]: Deleted in all componets

File: mw-webapp/src/component/table/LoadingContext.tsx (hunk 7 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/table/LoadingContext.tsx]: Deleted in all componets

**Review by Howl404 — APPROVED** [review]

**Pull request merged**